databuilder/query_engines/base_sql.py

```diff
@@ -83,6 +83,10 @@ def select_patient_id_for_population(self, population_expression):
     def get_sql(self, node):
         assert False, f"Unhandled node: {node}"
 
+    @singledispatchmethod
+    def get_table(self, node):
+        assert False, f"Unhandled node: {node}"
+
```

Comment:
This seems reasonable. Did you consider renaming the other type to `get_column`?

Reply:
`get_column` wouldn't be _quite_ right in this case. Because `get_table` really does return a reference to a table, but the thing currently called `get_sql` returns an arbitrarily complex expression that can used in places a column reference would be.

`get_sql` is obviously not a great name, but I found it hard to think of the right term if we're going to use `get_expr` to mean "thing which is not a predicate". I think my grasp on the terminology has got a bit shaky. There might be better terms for all the things somewhere.